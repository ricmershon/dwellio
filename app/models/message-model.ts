import { Document, Schema, Types, model, models } from 'mongoose';
import { MessageInputType } from '@/app/schemas/message-schema';

export interface MessageDocument extends MessageInputType, Document {
    sender: Types.ObjectId,
    recipient: Types.ObjectId,
    property: Types.ObjectId | { _id: Types.ObjectId, name: string },
    read: boolean,
    createdAt: Date;
    updatedAt: Date;
}

const MessageSchema = new Schema({
    sender: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    recipient: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    property: { type: Schema.Types.ObjectId, ref: 'Property', required: true },
    name: { type: String, required: [true, 'Name is required'] },
    email: { type: String, required: [true, 'Email is required'] },
    phone: String,
    body: String,
    read: { type: Boolean, default: false }
}, {
    timestamps: true
});

const Message = models.Message || model<MessageDocument>('Message', MessageSchema);
export default Message;