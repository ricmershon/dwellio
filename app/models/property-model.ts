import { Document, Schema, Types, model, models } from 'mongoose';

import { PropertyImageData } from '@/app/lib/definitions';
import { PropertyInputType } from '@/app/schemas/property-schema';

export interface PropertyDocument extends Omit<PropertyInputType, 'imagesData'>, Document {
    owner: Types.ObjectId;
    isFeatured?: boolean;
    imagesData?: PropertyImageData[];
    createdAt: Date;
    updatedAt: Date;
}

const ImageSchema = new Schema({
    secureUrl: { type: String, required: true },
    publicId: { type: String, required: true }
}, { _id: false });

const PropertySchema = new Schema<PropertyDocument>({
    owner: { type: Schema.Types.ObjectId, required: true, ref: 'User' },
    name: { type: String, required: true },
    type: { type: String, required: true },
    description: { type: String  },
    location: {
        street: String,
        city: String,
        state: String,
        zipcode: String
    },
    beds: { type: Number, required: true },
    baths: { type: Number, required: true },
    squareFeet: { type: Number, required: true },
    amenities: [{ type: String }],
    rates: {
        nightly: Number,
        weekly: Number,
        monthly: Number
    },
    sellerInfo: {
        name: String,
        email: String,
        phone: String
    },
    imagesData: {
        type: [ImageSchema],
        required: true,
        validate: {
            validator: (value: PropertyImageData[]) => Array.isArray(value) && value.length > 0,
            message: 'At least one image is required in imagesData.',
        },
    },
    isFeatured: { type: Boolean, default: false }
}, {
    timestamps: true
});

const Property = models.Property || model<PropertyDocument>('Property', PropertySchema);
export default Property;