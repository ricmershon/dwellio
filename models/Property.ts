import { Document, Schema, Types, model, models } from 'mongoose';
import { UserInterface } from './User';

export interface PropertyInterface extends Document {
    _id: Types.ObjectId;
    owner: UserInterface;
    name: string;
    type: string;
    description?: string;
    location: {
        street: string;
        city: string;
        state: string;
        zipcode: string
    };
    beds: number;
    baths: number;
    square_feet: number;
    amenities?: Array<string>;
    rates: {
        nightly?: number;
        weekly?: number;
        monthly?: number
    };
    seller_info: {
        name: string;
        email: string;
        phone: string
    };
    images: Array<string>;
    is_featured: boolean;
    createdAt: Date;
    updatedAt: Date;
}

const PropertySchema = new Schema({
    _id: Schema.Types.ObjectId,
    owner: { type: Schema.Types.ObjectId, required: true, ref: 'User' },
    name: { type: String, required: true },
    type: { type: String, required: true },
    description: { type: String  },
    location: {
        street: String,
        city: String,
        state: String,
        zipcode: String
    },
    beds: { type: Number, required: true },
    baths: { type: Number, required: true },
    square_feet: { type: Number, required: true },
    amenities: [{ type: String }],
    rates: {
        nightly: Number,
        weekly: Number,
        monthly: Number
    },
    seller_info: {
        name: String,
        email: String,
        phone: String
    },
    images: [{ type: String }],
    is_featured: { type: Boolean, default: false }
}, {
    timestamps: true
});

// export const Property = models.Property || model('Property', PropertySchema);

export const Property = models.Property || model<PropertyInterface>('Property', PropertySchema);