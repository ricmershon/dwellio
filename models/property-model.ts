import { Document, Schema, Types, model, models } from "mongoose";

import { PropertyImageData } from "@/types/types";
import { PropertyInputType } from "@/schemas/property-schema";

export interface PropertyDocument extends Omit<PropertyInputType, "imagesData">, Document {
    owner: Types.ObjectId;
    isFeatured?: boolean;
    imagesData?: PropertyImageData[];
    createdAt: Date;
    updatedAt: Date;
}

const ImageSchema = new Schema({
    secureUrl: { type: String, required: true },
    publicId: { type: String, required: true },
    height: { type: Number, required: true },
    width: { type: Number, required: true },
}, { _id: false });

const PropertySchema = new Schema<PropertyDocument>({
    owner: { type: Schema.Types.ObjectId, required: true, ref: "User" },
    name: { type: String, required: true },
    type: { type: String, required: true },
    description: { type: String  },
    location: {
        street: String,
        city: String,
        state: String,
        zipcode: String
    },
    beds: { type: Number, required: true },
    baths: { type: Number, required: true },
    squareFeet: { type: Number, required: true },
    amenities: [{ type: String }],
    rates: {
        nightly: Number,
        weekly: Number,
        monthly: Number
    },
    sellerInfo: {
        name: String,
        email: String,
        phone: String
    },
    imagesData: {
        type: [ImageSchema],
        required: true,
        validate: {
            validator: (value: PropertyImageData[]) => Array.isArray(value) && value.length > 2,
            message: "At least three images are required.",
        },
    },
    isFeatured: { type: Boolean, default: false }
}, {
    timestamps: true
});

const Property = models.Property || model<PropertyDocument>("Property", PropertySchema);
export default Property;