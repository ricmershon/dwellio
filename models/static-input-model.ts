import { Schema, Document, model, models } from 'mongoose';

interface Amenity {
    id: string;
    value: string;
}

export interface StaticInputInterface extends Document {
    amenities: Amenity[];
    property_types: string[];
}

const AmenitySchema = new Schema<Amenity>({
    id: { type: String, required: true },
    value: { type: String, required: true }
}, {
    _id: false
});

const StaticInputSchema = new Schema<StaticInputInterface>({
    amenities: {
        type: [AmenitySchema],
        required: true
    },
    property_types: {
        type: [String],
        required: true
    }
});

const StaticInput = models.StaticInput || model<StaticInputInterface>('StaticInput', StaticInputSchema);
export default StaticInput;