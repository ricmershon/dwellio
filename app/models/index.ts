import Property, { PropertyInterface } from "@/app/models/property-model";
import User, { UserInterface} from "@/app/models/user-model";

export { Property, User };
export type { PropertyInterface, UserInterface }