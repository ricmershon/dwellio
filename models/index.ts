import Property, { PropertyDocument } from "@/models/property-model";
import User, { UserDocument} from "@/models/user-model";
import Message, { MessageDocument } from "@/models/message-model";
import StaticInputs, { StaticInputsDocument } from "@/models/static-inputs-model";

export { Property, User, Message, StaticInputs };
export type { PropertyDocument, UserDocument, MessageDocument, StaticInputsDocument }