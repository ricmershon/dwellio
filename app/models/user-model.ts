import { Document, Schema, Types, model, models } from 'mongoose';

import { PropertyDocument } from '@/app/models/property-document';

export interface UserDocument extends Document {
    username: string;
    email: string;
    image: string;
    favorites: Types.ObjectId[] & PropertyDocument[]
}

const UserSchema = new Schema({
    username: { type: String, required: [true, 'Username is required'], unique: true },
    email: {
        type: String,
        unique: [true, 'Email already exists'],
        required: [true, 'Email is required']
    },
    image: { type: String },
    favorites: [{ type: Schema.Types.ObjectId, ref: 'Property' }]
}, {
    timestamps: true
});

const User = models.User || model<UserDocument>('User', UserSchema);
export default User;