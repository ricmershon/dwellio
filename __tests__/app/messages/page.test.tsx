// Mock all external dependencies first
jest.mock('@/lib/data/message-data', () => ({
    fetchMessages: jest.fn()
}));

jest.mock('@/utils/require-session-user', () => ({
    requireSessionUser: jest.fn()
}));

jest.mock('@/ui/messages/message-card', () => ({
    __esModule: true,
    default: ({ message }: { message: any }) => (
        <div data-testid="message-card" data-message-id={message._id}>
            Message Card - {message.body}
        </div>
    ),
}));

jest.mock('@/ui/shared/breadcrumbs', () => ({
    __esModule: true,
    default: ({ breadcrumbs }: { breadcrumbs: any[] }) => (
        <nav data-testid="breadcrumbs">
            {breadcrumbs.map((breadcrumb, index) => (
                <span key={index} data-active={breadcrumb.active || false}>
                    {breadcrumb.label}
                </span>
            ))}
        </nav>
    ),
}));

// Import after mocks
import React from 'react';
import { render, screen } from '@/__tests__/test-utils';
import MessagesPage from '@/app/messages/page';
import { MessageDocument } from '@/models';

const { fetchMessages } = jest.requireMock('@/lib/data/message-data');
const { requireSessionUser } = jest.requireMock('@/utils/require-session-user');

describe('MessagesPage', () => {
    const mockSessionUser = {
        id: '507f1f77bcf86cd799439011',
        email: 'test@example.com',
        name: 'Test User'
    };

    const mockMessages = [
        {
            _id: 'msg1',
            body: 'Test message 1',
            sender: { username: 'sender1' },
            property: { name: 'Property 1' },
            recipient: mockSessionUser.id,
            read: false,
            createdAt: new Date('2024-01-01')
        },
        {
            _id: 'msg2',
            body: 'Test message 2',
            sender: { username: 'sender2' },
            property: { name: 'Property 2' },
            recipient: mockSessionUser.id,
            read: true,
            createdAt: new Date('2024-01-02')
        },
        {
            _id: 'msg3',
            body: 'Test message 3',
            sender: { username: 'sender3' },
            property: { name: 'Property 3' },
            recipient: mockSessionUser.id,
            read: false,
            createdAt: new Date('2024-01-03')
        }
    ] as unknown as MessageDocument[];

    beforeEach(() => {
        jest.clearAllMocks();
        requireSessionUser.mockResolvedValue(mockSessionUser);
        fetchMessages.mockResolvedValue(mockMessages);
    });

    describe('Authentication Integration', () => {
        it('should call requireSessionUser before rendering', async () => {
            const component = await MessagesPage();
            render(component);

            expect(requireSessionUser).toHaveBeenCalledTimes(1);
        });

        it('should pass user ID to data fetching', async () => {
            const component = await MessagesPage();
            render(component);

            expect(fetchMessages).toHaveBeenCalledWith(mockSessionUser.id);
        });

        it('should handle authentication redirects', async () => {
            requireSessionUser.mockRejectedValue(new Error('Authentication required'));

            await expect(MessagesPage()).rejects.toThrow('Authentication required');
        });

        it('should handle session errors gracefully', async () => {
            requireSessionUser.mockRejectedValue(new Error('Session expired'));

            await expect(MessagesPage()).rejects.toThrow('Session expired');
        });
    });

    describe('Data Fetching Integration', () => {
        it('should call fetchMessages with user ID', async () => {
            const component = await MessagesPage();
            render(component);

            expect(fetchMessages).toHaveBeenCalledWith(mockSessionUser.id);
            expect(fetchMessages).toHaveBeenCalledTimes(1);
        });

        it('should handle successful data retrieval', async () => {
            const component = await MessagesPage();
            render(component);

            const messageCards = screen.getAllByTestId('message-card');
            expect(messageCards).toHaveLength(3);
        });

        it('should process message array correctly', async () => {
            const component = await MessagesPage();
            render(component);

            expect(screen.getByText('Message Card - Test message 1')).toBeInTheDocument();
            expect(screen.getByText('Message Card - Test message 2')).toBeInTheDocument();
            expect(screen.getByText('Message Card - Test message 3')).toBeInTheDocument();
        });

        it('should handle empty message arrays', async () => {
            fetchMessages.mockResolvedValue([]);

            const component = await MessagesPage();
            render(component);

            expect(screen.getByText('You have no messages.')).toBeInTheDocument();
            expect(screen.queryAllByTestId('message-card')).toHaveLength(0);
        });
    });

    describe('Component Orchestration', () => {
        it('should render main element as container', async () => {
            const component = await MessagesPage();
            render(component);

            const mainElement = screen.getByRole('main');
            expect(mainElement).toBeInTheDocument();
        });

        it('should render Breadcrumbs with correct props', async () => {
            const component = await MessagesPage();
            render(component);

            const breadcrumbs = screen.getByTestId('breadcrumbs');
            expect(breadcrumbs).toBeInTheDocument();
            expect(screen.getByText('Home')).toBeInTheDocument();
            expect(screen.getByText('Your Messages')).toBeInTheDocument();
        });

        it('should render MessageCard for each message', async () => {
            const component = await MessagesPage();
            render(component);

            const messageCards = screen.getAllByTestId('message-card');
            expect(messageCards).toHaveLength(3);

            expect(messageCards[0]).toHaveAttribute('data-message-id', 'msg1');
            expect(messageCards[1]).toHaveAttribute('data-message-id', 'msg2');
            expect(messageCards[2]).toHaveAttribute('data-message-id', 'msg3');
        });

        it('should maintain proper component hierarchy', async () => {
            const component = await MessagesPage();
            render(component);

            const main = screen.getByRole('main');
            const breadcrumbs = screen.getByTestId('breadcrumbs');
            const messageCards = screen.getAllByTestId('message-card');

            expect(main).toContainElement(breadcrumbs);
            messageCards.forEach(card => {
                expect(main).toContainElement(card);
            });
        });
    });

    describe('Breadcrumbs Integration', () => {
        it('should pass correct breadcrumb array', async () => {
            const component = await MessagesPage();
            render(component);

            const breadcrumbs = screen.getByTestId('breadcrumbs');
            expect(breadcrumbs).toBeInTheDocument();
            expect(screen.getByText('Home')).toBeInTheDocument();
            expect(screen.getByText('Your Messages')).toBeInTheDocument();
        });

        it('should set "Your Messages" as active', async () => {
            const component = await MessagesPage();
            render(component);

            const activeElement = screen.getByText('Your Messages');
            expect(activeElement).toHaveAttribute('data-active', 'true');
        });

        it('should include Home link', async () => {
            const component = await MessagesPage();
            render(component);

            const homeElement = screen.getByText('Home');
            expect(homeElement).toHaveAttribute('data-active', 'false');
        });

        it('should handle navigation properly', async () => {
            const component = await MessagesPage();
            render(component);

            const breadcrumbs = screen.getByTestId('breadcrumbs');
            expect(breadcrumbs).toBeInTheDocument();
            expect(breadcrumbs.children).toHaveLength(2);
        });
    });

    describe('Message List Rendering', () => {
        it('should iterate over messages array', async () => {
            const component = await MessagesPage();
            render(component);

            const messageCards = screen.getAllByTestId('message-card');
            expect(messageCards).toHaveLength(mockMessages.length);
        });

        it('should use message._id as React key', async () => {
            const component = await MessagesPage();
            render(component);

            const messageCards = screen.getAllByTestId('message-card');
            expect(messageCards[0]).toHaveAttribute('data-message-id', 'msg1');
            expect(messageCards[1]).toHaveAttribute('data-message-id', 'msg2');
            expect(messageCards[2]).toHaveAttribute('data-message-id', 'msg3');
        });

        it('should pass complete message object', async () => {
            const component = await MessagesPage();
            render(component);

            expect(screen.getByText('Message Card - Test message 1')).toBeInTheDocument();
            expect(screen.getByText('Message Card - Test message 2')).toBeInTheDocument();
            expect(screen.getByText('Message Card - Test message 3')).toBeInTheDocument();
        });

        it('should maintain message order', async () => {
            const component = await MessagesPage();
            render(component);

            const messageCards = screen.getAllByTestId('message-card');
            const messageIds = messageCards.map(card => card.getAttribute('data-message-id'));
            expect(messageIds).toEqual(['msg1', 'msg2', 'msg3']);
        });
    });

    describe('Empty State Handling', () => {
        it('should show "You have no messages." when empty', async () => {
            fetchMessages.mockResolvedValue([]);

            const component = await MessagesPage();
            render(component);

            expect(screen.getByText('You have no messages.')).toBeInTheDocument();
        });

        it('should hide message list when no messages', async () => {
            fetchMessages.mockResolvedValue([]);

            const component = await MessagesPage();
            render(component);

            expect(screen.queryAllByTestId('message-card')).toHaveLength(0);
        });

        it('should maintain layout structure', async () => {
            fetchMessages.mockResolvedValue([]);

            const component = await MessagesPage();
            render(component);

            const main = screen.getByRole('main');
            const breadcrumbs = screen.getByTestId('breadcrumbs');
            
            expect(main).toBeInTheDocument();
            expect(breadcrumbs).toBeInTheDocument();
            expect(screen.getByText('You have no messages.')).toBeInTheDocument();
        });

        it('should handle null arrays', async () => {
            fetchMessages.mockResolvedValue(null);

            // The component doesn't handle null arrays - it will throw
            await expect(MessagesPage()).rejects.toThrow("Cannot read properties of null (reading 'length')");
        });
    });

    describe('Layout and Styling', () => {
        it('should apply margin auto to container', async () => {
            const component = await MessagesPage();
            render(component);

            const container = screen.getByRole('main').querySelector('.m-auto');
            expect(container).toBeInTheDocument();
        });

        it('should use space-y-5 for message spacing', async () => {
            const component = await MessagesPage();
            render(component);

            const messageContainer = screen.getByRole('main').querySelector('.space-y-5');
            expect(messageContainer).toBeInTheDocument();
        });

        it('should maintain responsive design', async () => {
            const component = await MessagesPage();
            render(component);

            const main = screen.getByRole('main');
            expect(main).toBeInTheDocument();
            
            const container = main.querySelector('.m-auto');
            expect(container).toBeInTheDocument();
            
            const messageArea = container?.querySelector('.space-y-5');
            expect(messageArea).toBeInTheDocument();
        });

        it('should handle overflow content', async () => {
            const manyMessages = Array.from({ length: 20 }, (_, i) => ({
                _id: `msg${i}`,
                body: `Test message ${i}`,
                sender: { username: `sender${i}` },
                property: { name: `Property ${i}` },
                recipient: mockSessionUser.id,
                read: i % 2 === 0,
                createdAt: new Date(`2024-01-${String(i + 1).padStart(2, '0')}`)
            })) as unknown as MessageDocument[];

            fetchMessages.mockResolvedValue(manyMessages);

            const component = await MessagesPage();
            render(component);

            const messageCards = screen.getAllByTestId('message-card');
            expect(messageCards).toHaveLength(20);
        });
    });

    describe('Error Handling and Edge Cases', () => {
        it('should handle data fetch errors', async () => {
            fetchMessages.mockRejectedValue(new Error('Database connection failed'));

            await expect(MessagesPage()).rejects.toThrow('Database connection failed');
        });

        it('should handle malformed session data', async () => {
            requireSessionUser.mockResolvedValue({ id: null });

            // Should handle null user ID - fetchMessages will be called with null
            const component = await MessagesPage();
            render(component);

            expect(fetchMessages).toHaveBeenCalledWith(null);
        });

        it('should handle missing user id in session', async () => {
            requireSessionUser.mockResolvedValue({});

            // Should handle missing id field
            const component = await MessagesPage();
            render(component);

            expect(fetchMessages).toHaveBeenCalledWith(undefined);
        });

        it('should handle message data transformation errors', async () => {
            const malformedMessages = [
                { _id: 'msg1' }, // Missing required fields
                { _id: 'msg2', body: 'Test' } // Partial data
            ] as unknown as MessageDocument[];

            fetchMessages.mockResolvedValue(malformedMessages);

            const component = await MessagesPage();
            render(component);

            const messageCards = screen.getAllByTestId('message-card');
            expect(messageCards).toHaveLength(2);
        });

        it('should maintain security boundaries', async () => {
            requireSessionUser.mockRejectedValue(new Error('Unauthorized'));

            await expect(MessagesPage()).rejects.toThrow('Unauthorized');
            expect(fetchMessages).not.toHaveBeenCalled();
        });
    });

    describe('Server Component Behavior', () => {
        it('should be an async server component', async () => {
            expect(MessagesPage).toBeDefined();
            
            const result = MessagesPage();
            expect(result).toBeInstanceOf(Promise);
        });

        it('should handle server-side data fetching', async () => {
            const component = await MessagesPage();
            render(component);

            // Both auth and data fetch should be called server-side
            expect(requireSessionUser).toHaveBeenCalled();
            expect(fetchMessages).toHaveBeenCalled();
        });

        it('should process data before rendering', async () => {
            const component = await MessagesPage();
            render(component);

            // Data should be pre-processed and ready for rendering
            expect(screen.getAllByTestId('message-card')).toHaveLength(3);
            expect(screen.getByTestId('breadcrumbs')).toBeInTheDocument();
        });
    });
});