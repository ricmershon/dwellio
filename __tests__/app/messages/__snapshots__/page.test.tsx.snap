// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`MessagesPage Snapshots should match snapshot for breadcrumbs integration: breadcrumbs-integration 1`] = `
<main>
  <nav
    data-testid="breadcrumbs"
  >
    <span
      data-active="false"
    >
      Home
    </span>
    <span
      data-active="true"
    >
      Your Messages
    </span>
  </nav>
  <div
    class="m-auto"
  >
    <div
      class="space-y-5"
    >
      <p>
        You have no messages.
      </p>
    </div>
  </div>
</main>
`;

exports[`MessagesPage Snapshots should match snapshot for empty messages state: empty-messages-state 1`] = `
<main>
  <nav
    data-testid="breadcrumbs"
  >
    <span
      data-active="false"
    >
      Home
    </span>
    <span
      data-active="true"
    >
      Your Messages
    </span>
  </nav>
  <div
    class="m-auto"
  >
    <div
      class="space-y-5"
    >
      <p>
        You have no messages.
      </p>
    </div>
  </div>
</main>
`;

exports[`MessagesPage Snapshots should match snapshot for long message list: long-message-list 1`] = `
<main>
  <nav
    data-testid="breadcrumbs"
  >
    <span
      data-active="false"
    >
      Home
    </span>
    <span
      data-active="true"
    >
      Your Messages
    </span>
  </nav>
  <div
    class="m-auto"
  >
    <div
      class="space-y-5"
    >
      <div
        data-message-id="long-msg-0"
        data-testid="message-card"
      >
        Message Card - 
        Long message 0 with detailed content about property inquiry and requirements
      </div>
      <div
        data-message-id="long-msg-1"
        data-testid="message-card"
      >
        Message Card - 
        Long message 1 with detailed content about property inquiry and requirements
      </div>
      <div
        data-message-id="long-msg-2"
        data-testid="message-card"
      >
        Message Card - 
        Long message 2 with detailed content about property inquiry and requirements
      </div>
      <div
        data-message-id="long-msg-3"
        data-testid="message-card"
      >
        Message Card - 
        Long message 3 with detailed content about property inquiry and requirements
      </div>
      <div
        data-message-id="long-msg-4"
        data-testid="message-card"
      >
        Message Card - 
        Long message 4 with detailed content about property inquiry and requirements
      </div>
      <div
        data-message-id="long-msg-5"
        data-testid="message-card"
      >
        Message Card - 
        Long message 5 with detailed content about property inquiry and requirements
      </div>
      <div
        data-message-id="long-msg-6"
        data-testid="message-card"
      >
        Message Card - 
        Long message 6 with detailed content about property inquiry and requirements
      </div>
      <div
        data-message-id="long-msg-7"
        data-testid="message-card"
      >
        Message Card - 
        Long message 7 with detailed content about property inquiry and requirements
      </div>
      <div
        data-message-id="long-msg-8"
        data-testid="message-card"
      >
        Message Card - 
        Long message 8 with detailed content about property inquiry and requirements
      </div>
      <div
        data-message-id="long-msg-9"
        data-testid="message-card"
      >
        Message Card - 
        Long message 9 with detailed content about property inquiry and requirements
      </div>
    </div>
  </div>
</main>
`;

exports[`MessagesPage Snapshots should match snapshot for multiple messages (mixed read/unread): multiple-mixed-messages 1`] = `
<main>
  <nav
    data-testid="breadcrumbs"
  >
    <span
      data-active="false"
    >
      Home
    </span>
    <span
      data-active="true"
    >
      Your Messages
    </span>
  </nav>
  <div
    class="m-auto"
  >
    <div
      class="space-y-5"
    >
      <div
        data-message-id="msg1"
        data-testid="message-card"
      >
        Message Card - 
        Test message 1
      </div>
      <div
        data-message-id="msg2"
        data-testid="message-card"
      >
        Message Card - 
        Test message 2
      </div>
      <div
        data-message-id="msg3"
        data-testid="message-card"
      >
        Message Card - 
        Test message 3
      </div>
    </div>
  </div>
</main>
`;

exports[`MessagesPage Snapshots should match snapshot for page layout structure: page-layout-structure 1`] = `
<main>
  <nav
    data-testid="breadcrumbs"
  >
    <span
      data-active="false"
    >
      Home
    </span>
    <span
      data-active="true"
    >
      Your Messages
    </span>
  </nav>
  <div
    class="m-auto"
  >
    <div
      class="space-y-5"
    >
      <div
        data-message-id="msg1"
        data-testid="message-card"
      >
        Message Card - 
        Test message 1
      </div>
      <div
        data-message-id="msg2"
        data-testid="message-card"
      >
        Message Card - 
        Test message 2
      </div>
      <div
        data-message-id="msg3"
        data-testid="message-card"
      >
        Message Card - 
        Test message 3
      </div>
    </div>
  </div>
</main>
`;

exports[`MessagesPage Snapshots should match snapshot for single message display: single-message-display 1`] = `
<main>
  <nav
    data-testid="breadcrumbs"
  >
    <span
      data-active="false"
    >
      Home
    </span>
    <span
      data-active="true"
    >
      Your Messages
    </span>
  </nav>
  <div
    class="m-auto"
  >
    <div
      class="space-y-5"
    >
      <div
        data-message-id="msg1"
        data-testid="message-card"
      >
        Message Card - 
        Test message 1
      </div>
    </div>
  </div>
</main>
`;
